src/compiler/transformers/ts.ts

```diff
@@ -2872,6 +2871,30 @@ namespace ts {
                 /*location*/ blockLocation,
                 /*multiLine*/ true
             );
+
+            // namespace hello.hi.world {
+            //      function foo() {}
+            //
+            //      // TODO, blah
+            // }
+            //
+            // should be emitted as
+            //
+            // var hello;
+            // (function (hello) {
+            //     var hi;
+            //     (function (hi) {
+            //         var world;
+            //         (function (world) {
+            //             function foo() { }
+            //             // TODO, blah
+            //         })(world = hi.world || (hi.world = {}));
+            //     })(hi = hello.hi || (hello.hi = {}));
+            // })(hello || (hello = {}));
+            // so if the block is a transformed module declaration, turn off the comment emit

```

Comment:
I think what you mean to clarify is that this prevents emitting the same comment for every implicitly nested namespace block.